### PR TITLE
Correction in the line 309 of the azure_rm.py script

### PR DESCRIPTION
**Pull request by AlexonOliveiraRH**
Fetching dynamic inventory throws error : 'Cloud' object has no attribute 'decode'.

To solve this, the following change was made in the line 309:

From:
matched_clouds = [x for x in all_clouds if x.name == raw_cloud_env]
To:
matched_clouds = [x for x in all_clouds if x.name == raw_cloud_env.name]

Reference:
https://github.com/Azure/azure-sdk-for-python/issues/3009

PS. This change was tested in a production environment successfully.

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: contrib/inventory/azure_rm.py
Comment (yuwzho) [inline contrib/inventory/azure_rm.py:309]: Please also replace the `raw_cloud_env` this in the following context

**Review by yuwzho — CHANGES_REQUESTED** [review]

**Comment by Fred-sun** [comment]
@AlexonOliveiraRH Could you can give a fixes according by the above comments? Thank you very much!

**Comment by AlexonOliveiraRH** [comment]
@Fred-sun, sorry, didn't get you. I did suggest the fix for that as commented by @yuwzho, I did apply it to my fork but I don't have write access to the branch to merge it.

File: contrib/inventory/azure_rm.py
Comment (AlexonOliveiraRH) [reply contrib/inventory/azure_rm.py:309]: @yuwzho I already applied that change in my fork, tested in the customer environment successfully and I'm just waiting for the approval to merge it to the branch. Thank you.

**Comment by Fred-sun** [comment]
@yuwzho @yungezz Please help review this PR and help merge it when you're free! Thank you very much!

**Comment by MyronFanQiu** [comment]
@AlexonOliveiraRH Hi~ If you have fixed the comments, please re-push the newest branch to your git repo. The PR doesn't update for now.

**Comment by Fred-sun** [comment]
@AlexonOliveiraRH Are you still following this PR? You can follow the comments and re-upload the latest changes to your branch, and if the review passes, the community maintainer will merge into the repo, thanks!

**Comment by AlexonOliveiraRH** [comment]
Hey guys, sorry for my delayed answer. @Fred-sun, yes, I'm still following the PR. I've already pushed the changes to my branch "patch-1" since the first time. Are you not being able to see it? Please, let me know if not.

**Comment by Fred-sun** [comment]
@AlexonOliveiraRH Sorry to reply late! There is no new update on AlexonOliveiraRH:patch-1, please double check you have commit successful. Thank you very much!

**Comment by AlexonOliveiraRH** [comment]
@Fred-sun, Sorry to reply late! I'm kinda busy these past weeks. I think I figured out what's going on: I've applied the suggested change via Git WebGUI and for some reason it wasn't working, I don't know why, but it was showing up right for me. So I did it all over again, this time via CLI and now it seems to be worked fine. Please check AlexonOliveiraRH:devel now to see the update. Thanks.

**Comment by Fred-sun** [comment]
@AlexonOliveiraRH You can merge the changes on the AlexonOliveiraRH:devel main branch into the patch-1 branch, which is convenient for review. If approved, it can be directly merged into the remote repo. Thanks a lot!

**Comment by AlexonOliveiraRH** [comment]
@Fred-sun, gotcha. Done! Please, check it now.

**Comment by Fred-sun** [comment]
@yungezz @yuwzho @zikalino Please take a look when you're free! Thanks a lot!

**Comment by Fred-sun** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@mybayern1974 Please take a review this PR when you're free（Change is very small)! Thanks a lot!

**Comment by Fred-sun** [comment]
ping to merge

**Comment by Fred-sun** [comment]
@mybayern1974 @haiyuazhang review_need!

**Comment by Fred-sun** [comment]
ping to merge!

**Comment by Fred-sun** [comment]
@haiyuazhang @gavinfish The change is very small, it should be able to be merged directly, please review. Thank you very much!

**Comment by Fred-sun** [comment]
@gavinfish @haiyuazhang review_need!

File: contrib/inventory/azure_rm.py
Comment (gavinfish) [inline contrib/inventory/azure_rm.py:312]: If the issue is corrected. There will be some cases triggering this branch, please also update the user cases of the `raw_cloud_env` variable.

**Comment by Fred-sun** [comment]
@AlexonOliveiraRH Thank you for taking the time to contribute to this PR. We will transfer ansible 's azure module related Issue and PR to azure collection (https://github.com/ansible-collections/azure/pulls), can you transfer the Issue to azure collection repo?

**Pull request closed without merging**
